lib/internal/module.js

```diff
@@ -3,11 +3,34 @@
 exports = module.exports = {
   makeRequireFunction,
   stripBOM,
-  addBuiltinLibsToObject
+  addBuiltinLibsToObject,
+  applyTransforms
 };
 
 exports.requireDepth = 0;
 
+const transforms = [];
+
+function addTransform(transform) {
+  transforms.push(transform)
+}
+function removeTransform(transform) {
+  const index = transforms.indexOf(transform)
+  if (index >= 0) {
+    transforms.splice(index, 1)
+  }
+}
+
+function applyTransforms(content, filename) {
+  return transforms.reduce((content, transform) => {
+    const next = transform(content, filename);
+    if (typeof next !== 'string') {
+      throw new Error('Module transforms must return the modified content')
+    }
+    return next;
+  }, content);
+}
+
```

Comment:
So you reimplemented `EventEmmiter` only with an explicit return. Do you really feel it's worth the duplication?
You could instead just wrap it, and win!
```js
var ee = new EventEmitter();
var tMap = new WeakMap();
function applyTransforms(content, filename) {
  var e = {content, filename};
  ee.emit('moduleLoading', e);
  return e.content;
}
function addTransform(transform) {
  var t = (e) => {e.content = transform(e.content, e.filename)};
  tmap.add(transform, t);
  ee.addListener('moduleLoading', t)
}
function removeTransform(transform) { ee.removeListener('moduleLoading', tMap.get(transform))}
```